BootloaderCorePkg/Library/SmbiosInitLib/SmbiosInitLib.c

```diff
@@ -379,11 +404,32 @@ AddSmbiosType (
 }
 
 
+/**
+  This function is called to initialize the SmbiosStringsPtr.
+**/
+VOID
+EFIAPI
+InitSmbiosStringPtr (
+  VOID
+  )
+{
+
+  EFI_STATUS          Status;
+  //
+  // Initialize SMBIOS String Ptr
+  //
+  Status = PcdSet32S (PcdSmbiosStringsPtr, (UINT32)DefaultSmbiosStrings);

```

Comment:
Since platform will need to override the string in the buffer,  so this PcdSmbiosStringPtr should be an allocated buffer.  We can copy the DefalutSmbiosStrings into the allcoated buffer.  Later on platform can update this buffer directly.

Reply:
Implemented the same and uploaded new patch for your review.